### PR TITLE
Use StatusBarWidgetFactory instead of deprecated StatusBar APIs

### PR DESCRIPTION
**Pull request by tumb1er**
StatusBar.addWidget API was removed in 2020.1.

This MR rewrites StatusBar widgets to use brand new StatusBarWidgetFactory.

PS StatusBarWidgetProvider is deprecated in 2020.1.

### CODE REVIEW

**Comment by tumb1er** [comment]
What I tested manually:
1. Detect missing git-flow binary (unsupported git-flow version in status bar with yes-no dialog)
2. Enable git-flow widget if git repository root is added on the fly
3. Disable git-flow widget if git repository root is deleted on the fly
4. Change "unsupported git-flow version" to normal widget after installing git-flow and enabling git repo

**Comment by tumb1er** [comment]
2019.1 does not compile, StatusBarWidgetFactory is missing.

**Comment by OpherV** [comment]
Merged. Thanks!

**Comment by tumb1er** [comment]
Great! Is there anything else in the code stopping from publish 2020.1 compatible release?

**Comment by OpherV** [comment]
Got a couple of loose ends to tie - planning to release the new version today

**Comment by OpherV** [comment]
unfortunately still stuck on the NPE coming from PR #295

File: src/main/java/gitflow/ui/GitflowWidget.java [hunk 1]
```diff
@@ -163,6 +203,8 @@ private void update() {
             emptyTextAndTooltip();
             return;
         }
+        checkSupportedVersion();
```
Comment (OpherV) [inline src/main/java/gitflow/ui/GitflowWidget.java:206]: This is problematic and might create performance issues if you're running this on EVERY git update

File: src/main/java/gitflow/ui/GitflowWidget.java [hunk 2]
```diff
@@ -46,16 +49,16 @@
...
 /**
  * Status bar widget which displays actions for git flow
  *
  * @author Kirill Likhodedov, Opher Vishnia, Alexander von Bremen-Kühne
  */
-public class GitflowWidget extends GitBranchWidget implements GitRepositoryChangeListener {
+public class GitflowWidget extends GitBranchWidget implements GitRepositoryChangeListener, StatusBarWidget.TextPresentation {
     private volatile String myText = "";
     private volatile String myTooltip = "";
+    private volatile boolean myGitFlowVersionSupported;
```
Comment (OpherV) [inline src/main/java/gitflow/ui/GitflowWidget.java:61]: No need to cache this here... it's already one version of Git Flow CLI per system and it's cached in GitflowVersionTester

**Pull request merged**